cheat.php

```diff
@@ -713,6 +775,55 @@ function ExecuteRequest( $Method, $URL, $Data = [] )
 	return $Data;
 }
 
+function GetLocalScriptHash( )
+{
+	return sha1_file( __FILE__ );
+}
+
+function GetRepositoryScriptHash( )
+{
+	$c_r = curl_init( );
+
+	curl_setopt_array( $c_r, [
+		CURLOPT_USERAGENT      => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3464.0 Safari/537.36',
+		CURLOPT_RETURNTRANSFER => true,
+		CURLOPT_ENCODING       => 'gzip',
+		CURLOPT_TIMEOUT        => 15,
+		CURLOPT_CONNECTTIMEOUT => 5,
+		CURLOPT_HEADER         => 1,
+		CURLOPT_CAINFO         => __DIR__ . '/cacert.pem',
+	] );
+
+	curl_setopt( $c_r, CURLOPT_URL, 'https://api.github.com/repos/SteamDatabase/SalienCheat/git/trees/master' );
+	curl_setopt( $c_r, CURLOPT_HTTPGET, 1 );
+
+	do
+	{
+		$Data = curl_exec( $c_r );
+
+		$HeaderSize = curl_getinfo( $c_r, CURLINFO_HEADER_SIZE );
+		$Data = substr( $Data, $HeaderSize );
+		$Data = json_decode( $Data, true );
+
+		if ( isset( $Data[ 'tree' ] ) )
+		{
+			foreach( $Data[ 'tree' ] as &$File )
+			{
+				if ( isset( $File[ 'path' ] ) && $File[ 'path' ] === "cheat.php" )
+				{
+					if ( isset( $File[ 'sha' ]) )
+						return $File[ 'sha' ];
+
+					break;
+				}
+			}
+		}
+
+		Msg( '{lightred}-- Failed to check for script in repository... Retrying' );
+	}
+	while( true && sleep( 1 ) === 0 );
```

Comment:
Remove this whole `do/while` logic, just execute the request once.

If GitHub fails we don't want to spam it, or block the script.